File: templates/typescript/src/greeting.ts
export type GreetingInput = {
  name: string;
};

export function createGreeting(input: GreetingInput): string {
  const name = input.name.trim();

  if (name.length === 0) {
    throw new Error("name must not be empty");
  }

  return `Hello, ${name}.`;
}

File: templates/go/internal/greeting/greeting.go
package greeting

import (
	"fmt"
	"strings"
)

type Input struct {
	Name string
}

func Create(input Input) (string, error) {
	name := strings.TrimSpace(input.Name)
	if name == "" {
		return "", fmt.Errorf("name must not be empty")
	}

	return fmt.Sprintf("Hello, %s.", name), nil
}
